QueueInteropTransport.php

```diff
@@ -140,17 +139,17 @@ public function send(Envelope $message): Envelope
         }
 
         try {
-            $producer->send($topic, $message);
+            $producer->send($topic, $interopMessage);
         } catch (InteropQueueException $e) {
             if ($this->contextManager->recoverException($e, $destination)) {
                 // The context manager recovered the exception, we re-try.
-                $this->send($message);
+                $envelope = $this->send($envelope);
```

Comment:
Note: is it really expected to throw below in case of recoverable exception & retry? 🤔 Or is this missing a `return` instead?

Reply:
this looks wrong indeed

Reply:
also there is possibly an endless loop if recoverException keeps returning true. maybe there should be a safe-guard counter.